File: g595.py
# -*- coding: utf-8 -*-
num = 0
n = int(input())
h = list(map(int, input().split()))
for index, value in enumerate(h):
    if (value == 0):
        if (index == 0):
            num += h[index+1]
        elif (index == n-1):
            num += h[index-1]
        else:
            if (h[index-1] < h[index+1]):
                num += h[index-1]
            else:
                num += h[index+1]
print(num)

File: g595.c
#include <stdio.h>

int main(void) {
    int n, h[100], num = 0;
    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &h[i]);
    }
    for (int i = 0; i < n; i++) {
        if (h[i] == 0) {
            if (i == 0) {
                num += h[i + 1];
            }
            else if (i == n - 1) {
                num += h[i - 1];
            }
            else {
                if (h[i - 1] < h[i + 1]) {
                    num += h[i - 1];
                }
                else {
                    num += h[i + 1];
                }
            }
        }
    }
    printf("%d\n", num);
    return 0; 
}